server/src/main/java/org/apache/cassandra/sidecar/routes/RouteBuilder.java

```diff
@@ -206,25 +214,29 @@ public static class Factory
         private final AuthorizationProvider authorizationProvider;
         private final AdminIdentityResolver adminIdentityResolver;
         private final AuthorizationParameterValidateHandler authZParameterValidateHandler;
+        private final SidecarMetrics sidecarMetrics;
 
         public Factory(AccessControlConfiguration accessControlConfiguration,
                        AuthorizationProvider authorizationProvider,
                        AdminIdentityResolver adminIdentityResolver,
-                       AuthorizationParameterValidateHandler authZParameterValidateHandler)
+                       AuthorizationParameterValidateHandler authZParameterValidateHandler,
+                       SidecarMetrics sidecarMetrics)
         {
 
             this.accessControlConfiguration = accessControlConfiguration;
             this.authorizationProvider = authorizationProvider;
             this.adminIdentityResolver = adminIdentityResolver;
             this.authZParameterValidateHandler = authZParameterValidateHandler;
+            this.sidecarMetrics = sidecarMetrics;
         }
 
         public RouteBuilder builderForRoute()
         {
             return new RouteBuilder(accessControlConfiguration,
                                     authorizationProvider,
                                     adminIdentityResolver,
-                                    authZParameterValidateHandler);
+                                    authZParameterValidateHandler,
```

Comment:
NIT, we should instead modify the ctor to take the Factory. i.e new RouteBuilder(this);